src/libPMacc/include/fields/operations/AddExchangeToBorder.hpp

```diff
@@ -0,0 +1,228 @@
+/* Copyright 2013-2017 Axel Huebl, Heiko Burau, Rene Widera, Marco Garten,
+ *                     Benjamin Worpitz
+ *
+ * This file is part of libPMacc.
+ *
+ * libPMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * libPMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with libPMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "pmacc_types.hpp"
+#include "mappings/kernel/MappingDescription.hpp"
+#include "mappings/kernel/ExchangeMapping.hpp"
+#include "fields/tasks/FieldFactory.hpp"
+#include "math/Vector.hpp"
+#include "traits/GetNumWorkers.hpp"
+
+#include <boost/core/ignore_unused.hpp>
+
+
+namespace PMacc
+{
+namespace fields
+{
+namespace operations
+{
+
+    /** add the intermediate to the local box
```

Comment:
Be less cryptic: *Add particles from a received temporary buffer (exchange) to the local particle box (border)*